### PR TITLE
editable snapshots [AS-578]

### PR DESCRIPTION
**Pull request by zarsky-broad**
Verified locally doing all kinds of crazy fun editing...

### CODE REVIEW

File: src/components/workspace-utils.js [hunk 1]
```diff
@@ -77,32 +100,104 @@ export const SnapshotInfo = ({ snapshotId, snapshotName }) => {
...
+          }, [icon('edit')])
+        ]),
+        div({ style: { ...Style.elements.sectionHeader, marginBottom: '0.2rem' } }, [
+          'Description:',
+          !editingDescription && h(Link, {
+            style: { marginLeft: '0.5rem' },
+            onClick: () => setNewDescription(description),
+            'aria-label': 'Edit description'
+          }, [icon('edit')])
+        ]),
```
Comment (davidangb) [inline src/components/workspace-utils.js:147]: I see the aria-labels for both name and description  ... could these also have hovertext/tooltips that indicate what they will do?

File: src/components/workspace-utils.js [hunk 2]
```diff
@@ -77,32 +100,104 @@ export const SnapshotInfo = ({ snapshotId, snapshotName }) => {
...
+          h(FormLabel, { htmlFor: id }, ['New snapshot name']),
+          h(TextInput, {
+            inputProps: {
+              id,
+              autoFocus: true,
+              placeholder: 'Enter a name',
+              value: newName,
+              onChange: setNewName
+            }
+          })
```
Comment (davidangb) [inline src/components/workspace-utils.js:214]: Add an aria-label - and avoid the errors in console!

File: src/components/workspace-utils.js [hunk 3]
```diff
@@ -77,32 +100,104 @@ export const SnapshotInfo = ({ snapshotId, snapshotName }) => {
...
+              renderingConfig: {
+                singleLineBreaks: false,
+                markedOptions: { sanitize: true, sanitizer: _.escape }
+              },
+              status: false
+            },
+            className: 'simplemde-container',
+            value: newDescription,
+            onChange: setNewDescription
+          }),
```
Comment (davidangb) [inline src/components/workspace-utils.js:162]: is there any config to suppress the deprecation warnings thrown by marked?

**Review by davidangb — CHANGES_REQUESTED** [review]
See comments inline; I am also unable to edit the name field, as the save button never enables.

File: src/components/workspace-utils.js (hunk 3 above)
Comment (zarsky-broad) [reply src/components/workspace-utils.js:162]: Unfortunately no - the newer versions of marked remove sanitizing and have you do it for yourself, which is why we haven't upgraded, as we never put in the time to find a sanitizer we liked...

Comment (davidangb) [reply src/components/workspace-utils.js:162]: yup, I understand why the warning exists, I was wondering if there's any way to turn off the logging for it so it doesn't show up in users' consoles

File: src/components/workspace-utils.js [hunk 4]
```diff
@@ -77,32 +100,104 @@ export const SnapshotInfo = ({ snapshotId, snapshotName }) => {
...
+      }, [
+        h(IdContainer, [id => h(Fragment, [
+          h(FormLabel, { htmlFor: id }, ['New snapshot name']),
+          h(TextInput, {
+            id,
+            autoFocus: true,
+            placeholder: 'Enter a name',
+            value: newName,
+            onChange: setNewName
+          })
```
Comment (davidangb) [inline src/components/workspace-utils.js:214]: 1. can you have this validate the input and disable the save button on an invalid name? Else it's far too easy for an end user to supply an illegal name and get an error on save. ("Name must be 1 to 63 alphanumeric characters or underscores, and cannot start with an underscore.")
2. errors on save appear not to be displayed - for me, they result in an infinite spinner with no feedback to the user.

**Review by davidangb — COMMENTED** [review]
code changes look good, but I have more functional requests, sorry ...

File: src/components/workspace-utils.js (hunk 3 above)
Comment (zarsky-broad) [reply src/components/workspace-utils.js:162]: ah, no, I don't believe so.

**Review by davidangb — APPROVED** [review]
:+1: looks good.

I found one corner case in manual testing - if you have two snapshots "foo" and "bar" in your workspace, the UI allows you to attempt to rename "foo" to "bar", which fails in the backend due to uniqueness. This allowed me to see that error-display was working well! It would be a nicety if the UI caught this, but I don't think it should block this PR.

**Comment by zarsky-broad** [comment]
> fails in the backend due to uniqueness

Yeah, that would require making the component aware of all of the other snapshot names, which is certainly doable, but I'm inclined to punt on it for now...

**Review by ansingh7115 — APPROVED** [review]
Tested locally - looks good to me!

**Pull request merged**